src/anyp/Uri.cc

```diff
@@ -471,24 +454,60 @@ AnyP::Uri::parse(const HttpRequestMethod& method, const char *url)
         }
     }
 
-    parseFinish(protocol, proto, urlpath, foundHost, SBuf(login), foundPort);
+    setScheme(scheme);
+    path(urlpath);
+    host(foundHost);
+    userInfo(SBuf(login));
+    port(foundPort);
     return true;
+
+    } catch (...) {
+        debugs(23, 2, "error: " << CurrentException << Raw("rawUrl", rawUrl.rawContent(), rawUrl.length()).minLevel(DBG_DATA));
```

Comment:
Please check whether minLevel() is needed here. Long Raw input will not be printed at lower debugging levels by default anyway, and it is probably a good idea to show short URIs for error messages.

Reply:
Done.